src/vs/workbench/parts/files/browser/fileActions.ts

```diff
@@ -1783,6 +1783,43 @@ export class OpenResourcesAction extends Action {
 	}
 }
 
+export class OpenRecentlyClosedResourcesAction extends Action {
+
+	public static ID = 'workbench.files.action.openRecentlyClosedFile';
+	public static LABEL = nls.localize('openRecentlyClosedFile', "Open Recently Closed File");
+
+	constructor(
+		id: string,
+		label: string,
+		@IPartService private partService: IPartService,
+		@IWorkbenchEditorService private editorService: IWorkbenchEditorService,
+		@IViewletService private viewletService: IViewletService,
+		@ITextFileService private textFileService: ITextFileService
+	) {
+		super(id, label);
+	}
+
+	public run(): TPromise<any> {
+		return this.partService.joinCreation().then(() => {
+			let viewletPromise = TPromise.as(null);
+			if (!this.partService.isSideBarHidden()) {
+				viewletPromise = this.viewletService.openViewlet(Files.VIEWLET_ID, false);
+			}
+
+			return viewletPromise.then(() => {
+				let resource = this.textFileService.getWorkingFilesModel().restoreRecentlyRemovedEntry();
+
```

Comment:
I suggest to change this method to something like "give me the last closed working file" and then handle everything in this action (add the file back to working files and open it).